control-base/src/main/java/fi/nls/oskari/control/data/MapLayerGroupsHandler.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    public void init() {
+        // setup service if it hasn't been initialized
+        if(oskariMapLayerGroupService == null) {
+            setOskariMapLayerGroupService(new OskariMapLayerGroupServiceIbatisImpl());
+        }
+    }
+
+    /**
+     * Handles listing and single theme find
+     * @param params
+     * @throws ActionException
+     */
+    public void handleGet(ActionParameters params) throws ActionException {
+        final int id = params.getHttpParam(PARAM_ID, -1);
+        if(id != -1) {
+            // find single group
+            final MaplayerGroup maplayerGroup = oskariMapLayerGroupService.find(id);
+            ResponseHelper.writeResponse(params, maplayerGroup.getAsJSON());
+            return;
+        }
+
+        // find all maplayerGroups
+        final List<MaplayerGroup> maplayerGroups = oskariMapLayerGroupService.findAll();
+        final JSONArray list = new JSONArray();
+        for (MaplayerGroup maplayerGroup : maplayerGroups) {
+            list.put(maplayerGroup.getAsJSON());
+        }
+        final JSONObject result = new JSONObject();
+        JSONHelper.putValue(result, "mapLayerGroups", list);
+        ResponseHelper.writeResponse(params, result);
+    }
+
+    /**
+     * Handles insert
+     * @param params
+     * @throws ActionException
+     */
+    public void handlePut(ActionParameters params) throws ActionException {
+        params.requireAdminUser();
+        MaplayerGroup maplayerGroup = populateFromRequest(params.getPayLoadJSON());
+        final int id = oskariMapLayerGroupService.insert(maplayerGroup);
+        // check insert by loading from DB
+        final MaplayerGroup savedMapLayerGroup = oskariMapLayerGroupService.find(id);
+        ResponseHelper.writeResponse(params, savedMapLayerGroup.getAsJSON());
+    }
+
+    /**
+     * Handles update
+     * @param params
+     * @throws ActionException
+     */
+    public void handlePost(ActionParameters params) throws ActionException {
+        params.requireAdminUser();
+        MaplayerGroup maplayerGroup = populateFromRequest(params.getPayLoadJSON());
+        maplayerGroup.setId(params.getRequiredParamInt(PARAM_ID));
+        oskariMapLayerGroupService.update(maplayerGroup);
+        ResponseHelper.writeResponse(params, maplayerGroup.getAsJSON());
+    }
+
+    /**
+     * Handles removal
+     * @param params
+     * @throws ActionException
+     */
+    public void handleDelete(ActionParameters params) throws ActionException {
+        params.requireAdminUser();
+        final int id = params.getRequiredParamInt(PARAM_ID);
+        final MaplayerGroup theme = oskariMapLayerGroupService.find(id);
+        final List<Integer> maplayerIds = oskariMapLayerGroupService.findMaplayersByGroup(id);
+        if(!maplayerIds.isEmpty()) {
+            // theme with maplayers under it can't be removed
+            throw new ActionParamsException("Maplayers linked to theme", JSONHelper.createJSONObject("code", "not_empty"));
```

Comment:
I think you should talk about groups and not themes on the comment/exception message.

Reply:
Rename the variable also (theme -> group)